spaghetto/README.md

```diff
@@ -148,4 +190,9 @@ workspace:
     output: "output"
     # optional, Boolean, fail the build if `spago.yml` has redundant/missing packages
     pedantic_packages: false
+    # optional, PsaOptions
+    # These options only take affect when
+    # a package is not selected (e.g. `spago build` vs `spago build -p pkgName`)
```

Comment:
I'm feeling weird about having a `build` config in the `package` section - we already have one here in the `workspace` section, and it contains some options already.

I think we should not have it in the `package` section at all, since a "build configuration" feels like a property of the overall build - I guess one could have different psa settings for different packages in the same workspace, but that sounds contrived to me: the usual situation in my experience is that one wants to censor some warning globally.

Reply:
A few comments about this:
- I only added a separate build option in the workspace spot because I wasn't sure how to deal with building the entire workspace (e.g. when no package is selected). Originally, it was just for each package directly.
- I can imagine some codebases wanting `--strict` to be required (e.g. core libraries) for their source globs but not for their test globs. Right now, Spago compiles both source and test code in the same run, which I find problematic because sometimes the src code is first updated and then the test code. Using the same set of PSA options for both globs is problematic. I imagine that will only be more so across packages.
- If the workspace option was provided, there are two ways to interpret it.
    1. What I've done in this PR: the workspace options are used when no packages are selected.
    2. What I'm hesitant to do because of readability: the workspace options could act as a "defaults" when everything else is otherwise unspecified. PSA options precedence would then be `CLI arg > package option > workspace option > spago PSA defaults`.
- Given a package `foo` which stores its source code in `foo/src/**/*.purs`, I may not want to put my test code in `foo/test/**/*.purs`. As an example, consider https://github.com/thomashoneyman/purescript-halogen-hooks/issues/24. The test code may be its own package suitable for downstream consumers. In order for Halogen Hooks to support that, it would need to define 2-3 packages:
    - src code for halogen-hooks
    - src code for halogen-hooks-test
    - test code that uses halogen-hooks-test to test halogen-hooks. I'm not sure if this would go in the `halogen-hooks-test`'s `test/**/*.purs` source globs or be its own not-to-be-published package that depends on the other two.